sympy/core/expr.py

```diff
@@ -680,15 +681,24 @@ def check_denominator_zeros(expression):
         # expression depends on them or not using differentiation. This is
         # not sufficient for all expressions, however, so we don't return
         # False if we get a derivative other than 0 with free symbols.
+        unk = False
         for w in wrt_number:
-            deriv = expr.diff(w)
+            try:
+                deriv = expr.diff(w)
+            except AttributeError:
+                # some arg within expr did not like to be differentiated
+                # e.g. issue # 25861
+                unk = True
+                continue
```

Comment:
Please don't catch AttributeError.

I have most than once tried to remove all places in the codebase that catch AttributeError (including very recently in gh-25753) but it keeps coming back.

The problem here is either:

- The call to diff should not raise AttributeError
- This code should not call diff naively like this.

Either way catching the exception is not the proper fix and `except AttributeError` should not even be used as a temporary workaround.

Reply:
3-arg KD now raises NotImplementedError

Reply:
What exactly is being differentiated that raises this error?

I think that either diff should not raise NotImplementedError or this code should not call diff.

NotImplementedError is better than AttributeError but still why are we in a situation where exceptions are possibly going to be raised at all?